### PR TITLE
feat: add ability to edit group gf-81

### PR DESCRIPTION
**Pull request by perekotypole**
- implemented backend ability to update groups data
- added options popover for table group's items and modal
- added update form
- solved problem with menu in table
- fixed bug `fix: The modal window is closed when the toast notification Error message appears on the Create group modal`

### CODE REVIEW

File: apps/frontend/src/assets/images/icons/edit.svg
Comment (kolibri753) [inline apps/frontend/src/assets/images/icons/edit.svg:1]: As I see, we eventually agreed to call this icon `pencil`

File: apps/frontend/src/assets/images/icons/options.svg
Comment (kolibri753) [inline apps/frontend/src/assets/images/icons/options.svg:1]: and this `ellipsis`

File: apps/frontend/src/assets/images/icons/edit.svg
Comment (perekotypole) [reply apps/frontend/src/assets/images/icons/edit.svg:1]: Yeap, I know. I'm just waiting for approval/merging and going to delete everything that's not needed.

File: apps/frontend/src/pages/access-management/libs/helpers/get-group-rows.helper.tsx
Comment (perekotypole) [inline apps/frontend/src/pages/access-management/libs/helpers/get-group-rows.helper.tsx:1]: @liza-veis I wonder if I will be beaten for the decision to change the helper format to TSX? I tried a lot, but this is the only solution that works as I need

Comment (liza-veis) [reply apps/frontend/src/pages/access-management/libs/helpers/get-group-rows.helper.tsx:1]: No, you won't, that's how it's supposed to be 😅. The only thing is that we need to move this column to getColumns helper, in getRows we only map data fields

File: apps/frontend/src/libs/hooks/use-popover-position/use-popover-position.hook.ts
Comment (kolibri753) [inline apps/frontend/src/libs/hooks/use-popover-position/use-popover-position.hook.ts:21]: I also added `const scrollOffset = window.scrollY || document.documentElement.scrollTop;` so the popover appears correctly positioned when the user has scrolled down the page.

File: apps/frontend/src/pages/access-management/access-management.tsx
Comment (perekotypole) [inline apps/frontend/src/pages/access-management/access-management.tsx:141]: @liza-veis About it https://github.com/BinaryStudioAcademy/bsa-2024-gitfit/pull/128#discussion_r1752149527
Is it okay?

File: apps/backend/src/modules/groups/group.controller.ts
Comment (GvoFor) [inline apps/backend/src/modules/groups/group.controller.ts:251]:
```suggestion
	 *          description: ID of the group to update
	 *          schema:
	 *            type: string
	 *      requestBody:
	 *        description: Updated group object
```

File: apps/frontend/src/libs/components/input/styles.module.css
Comment (GvoFor) [inline apps/frontend/src/libs/components/input/styles.module.css:61]: Common components shouldn't have margins
```suggestion
```

File: apps/frontend/src/pages/access-management/access-management.tsx
Comment (GvoFor) [inline apps/frontend/src/pages/access-management/access-management.tsx:190]:
```suggestion
			{hasGroupToEdit && (
```

File: apps/frontend/src/pages/access-management/libs/components/groups-table/groups-table.tsx
Comment (GvoFor) [inline apps/frontend/src/pages/access-management/libs/components/groups-table/groups-table.tsx:51]: what is the purpose of this `Fragment`?

File: apps/frontend/src/pages/access-management/libs/helpers/get-group-columns.helper.tsx
Comment (GvoFor) [inline apps/frontend/src/pages/access-management/libs/helpers/get-group-columns.helper.tsx:30]:
```suggestion
		id: "menu",
```
Why not?

File: apps/frontend/src/libs/components/input/styles.module.css
Comment (perekotypole) [reply apps/frontend/src/libs/components/input/styles.module.css:61]: I have to remove it, we no longer have `.input-checkbox`.

File: apps/frontend/src/pages/access-management/libs/components/groups-table/groups-table.tsx
Comment (perekotypole) [reply apps/frontend/src/pages/access-management/libs/components/groups-table/groups-table.tsx:51]: forgot to remove

File: apps/frontend/src/libs/hooks/use-popover-position/use-popover-position.hook.ts
Comment (what1s1ove) [inline apps/frontend/src/libs/hooks/use-popover-position/use-popover-position.hook.ts:11]: pass parameters as an object instead

**Review by what1s1ove — APPROVED** [review]

File: apps/frontend/src/pages/access-management/libs/components/groups-table/groups-table.tsx
Comment (liza-veis) [inline apps/frontend/src/pages/access-management/libs/components/groups-table/groups-table.tsx:36]: let's move it to a separate variable
```suggestion
		onEdit: handleEdit
```

File: apps/frontend/src/modules/groups/slices/group.slice.ts
Comment (liza-veis) [inline apps/frontend/src/modules/groups/slices/group.slice.ts:82]:
```suggestion
			state.groups = state.groups.map((group) =>
```

File: apps/frontend/src/pages/access-management/libs/types/group-actions.type.ts
Comment (liza-veis) [inline apps/frontend/src/pages/access-management/libs/types/group-actions.type.ts:1]: Remove this type, these handlers can be overwritten on each level you use this type

File: apps/frontend/src/pages/access-management/access-management.tsx
Comment (liza-veis) [inline apps/frontend/src/pages/access-management/access-management.tsx:136]: let's keep it near groupToEdit and rename it to groupToDeleteId to keep it consistent
```suggestion
	const [groupToDeleteId, setGroupToDeleteId] = useState<null | number>(null);
	const hasGroupToDelete = Boolean(groupToDeleteId);
```

File: apps/frontend/src/pages/access-management/access-management.tsx
Comment (liza-veis) [inline apps/frontend/src/pages/access-management/access-management.tsx:89]: handleLoadUsers / handleLoadGroups
```suggestion
	const handleLoadUsers = useCallback(() => {
```

File: apps/frontend/src/pages/access-management/access-management.tsx
Comment (liza-veis) [inline apps/frontend/src/pages/access-management/access-management.tsx:127]: handlers created inside component should be named with handle
```suggestion
	const handleEdit = useCallback(
```

File: apps/frontend/src/pages/access-management/access-management.tsx
Comment (liza-veis) [inline apps/frontend/src/pages/access-management/access-management.tsx:152]: instead of this helper let's add useEffect on totalItemsCount change in usePagination, and update onPageChange inside it to handle this boundary

```tsx
    useEffect(() => {
         onPageChange(page)
    }, [totalItemsCount])
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/access-management/access-management.tsx
Comment (perekotypole) [reply apps/frontend/src/pages/access-management/access-management.tsx]: I added it to `use-pagination.hook.ts`, and removed `setValidPage`
```
	const onPageChange = useCallback(
		(newPage: number) => {
			const totalPages = calculateTotalPages(pageSize, totalItemsCount);
			const validPage = Math.min(newPage, totalPages);

			setPage(validPage);
		},
		[page, pageSize, totalItemsCount],
	);

	useEffect(() => {
		onPageChange(page);
	}, [totalItemsCount]);
```
I hope I understood correctly. It seems to work

File: apps/frontend/src/libs/hooks/use-pagination/use-pagination.hook.ts
Comment (liza-veis) [inline apps/frontend/src/libs/hooks/use-pagination/use-pagination.hook.ts:63]: we also need to make sure it's more than initial page
```suggestion
			const validPage = Math.max(FIRST_PAGE, Math.min(newPage, totalPages));
```

File: apps/frontend/src/pages/access-management/access-management.tsx
Comment (liza-veis) [reply apps/frontend/src/pages/access-management/access-management.tsx]: yes, looks good to me

File: apps/frontend/src/libs/hooks/use-pagination/use-pagination.hook.ts
Comment (perekotypole) [reply apps/frontend/src/libs/hooks/use-pagination/use-pagination.hook.ts]: Oh sure, my bad!
I just saw this check in `calculateTotalPages` and didn't notice it's not related.

**Review by liza-veis — APPROVED** [review]

**Pull request merged**
